doc/source/train/user-guides/experiment-tracking.rst

```diff
@@ -1,195 +1,292 @@
-
-            .. code-block:: python
-                :emphasize-lines: 5,9
-                
-                import os
-                from pytorch_lightning.loggers.wandb import WandbLogger
-
-                # Grab the api key from the head node
-                WANDB_API_KEY = os.environ["WANDB_API_KEY"]
-
-                def train_func_per_worker():
-                    # Set environment variable on each worker
-                    os.environ["WANDB_API_KEY"] = WANDB_API_KEY
-                    wandb_logger = WandbLogger(...)
-                    ...
-
-                trainer = TorchTrainer(
-                    train_func_per_worker,
-                    ...
-                )
-        
-        .. tip::
+            from ray import train
+            import wandb
+
+            def train_func(config):
+                os.environ["WANDB_MODE"] = "offline"
+                if train.get_context().get_world_rank() == 0:
+                    wandb.init(
+                        dir="...",  # some shared storage path like "/mnt/cluster_storage"
+                        id=..., # or train.get_context().get_trial_id(),
+                        name=..., # or train.get_context().get_trial_name(),
+                        group=..., # or train.get_context().get_experiment_name(),
+                        # ...
+                    )
+
+                # ...
+
+                loss = optimize()
+
+                metrics = {"loss": loss}
+                # Only report the first worker results to wandb to avoid dup
+                if train.get_context().get_world_rank() == 0:
+                    wandb.log(metrics)
+
+                # ...
+
+                if train.get_context().get_world_rank() == 0:
+                    wandb.finish()
+
+
+    .. tab:: file based MLflow (offline)
+
+        .. code-block:: python
             
-            When performing **fault-tolerant training** with auto-restoration, be sure 
-            to specify a unique ID for the Loggers, so that the new workers report to
-            the same run after restoration.
+            from ray import train
+            import mlflow
+
+            # assuming passing a save dir through config
+            def train_func(config):
+                save_dir = config["save_dir"]
+                if train.get_context().get_world_rank() == 0:
+                    # mlflow works the best if this is a folder dedicated to mlruns.
+                    mlflow.set_tracking_uri(f"file:{save_dir}")
+                    mlflow.set_experiment("my_experiment")
+                    mlflow.start_run()
+
+                # ...
+
+                loss = optimize()
+
+                metrics = {"loss": loss}
+                # Only report the first worker results to mlflow to avoid dup
+                if train.get_context().get_world_rank() == 0:
+                    mlflow.log_metrics(metrics)
+
+    .. tab:: MLflow externally hosted by databricks (online)
```

Comment:
The tab names are too long and wrapped to the next line, can we make it shorter(and switch the offline/online order)? like

```
Wandb(online), Wandb(offline), MLflow (online), MLflow(offline) 
```
<img width="781" alt="Screenshot 2023-08-23 at 3 37 59 PM" src="https://github.com/ray-project/ray/assets/26745457/096d4da4-3469-4541-b017-d3e3dba521db">

Reply:
Done!